### DISCUSSION TITLE
Monitor open files

### DISCUSSION THREAD

**Discussion by cesarjorgemartinez** [nicolargo/glances#3227] [General]
#### Description

This utility can monitor open files number count (in real-time)?

Example count open files in the Linux machine, and count open files group by existing users.

#### Versions

Nothing

#### Logs

Nothing

**Comment by nicolargo** [comment]
Hi @cesarjorgemartinez 

you can use [AMP](https://glances.readthedocs.io/en/stable/aoa/amps.html) to do this.

Example to add to your Glances conf file:

```
[amp_countopenfile]
enable=true
regex=.*glances.*
refresh=30
one_line=true
command=/path/to/your/script.sh
```
With /path/to/your/script.sh:

```
lsof | wc -l
```

Do not forget to chmod +x /path/to/your/script.sh

**Comment by distributorofpain** [comment]
Thanks for the example!  Did this with a log file:
```
[amp_xmrig]
enable=true
regex=.*glances.*
refresh=30
one_line=false
command=/etc/glances/xmrigoutput.sh
```

```
#!/usr/bin/env python3
import os
os.system('tail -20 /etc/xmrig/xmrig.log | tac')
```
